src/contrib/inference/sghmc.jl

```diff
@@ -238,8 +238,8 @@ function AbstractMCMC.step(
     θ .+= (stepsize / 2) .* grad .+ sqrt(stepsize) .* randn(rng, eltype(θ), length(θ))
 
     # Save new variables and recompute log density.
-    vi[spl] = θ
-    model(rng, vi, spl)
+    vi = DynamicPPL.setindex!!(vi, θ, spl)
+    vi = last(DynamicPPL.evaluate!!(model, vi, DynamicPPL.SamplingContext(rng, spl)))
```

Comment:
Do we need a new logdensity function?

Reply:
And is sampling needed?

Reply:
Do we not need the `SamplingContext` in case this is used as part of a gibbs kernel?

Reply:
As in, in that case `spl` should only be dealing with a subset of the variables.

Reply:
Do we not have a sampler-specific evaluation context? It's mainly because we just evaluate (it seems) that I think passing around `rng` seems unintuitive.